packages/zilliqa-js-crypto/src/util.ts

```diff
@@ -199,3 +200,87 @@ export const verifyPrivateKey = (privateKey: string): boolean => {
   const { result } = keyPair.validate();
   return result;
 };
+
+/**
+ * getAddress
+ *
+ * assume there is an address
+ * and transform it to target `toType`,
+ * also use `fromTypes` to limit the types with `type AddressType` array
+ *
+ *
+ * @param {string} address - an address to transform
+ * @param {AddressType} toType - to target `type AddressType`, if undefined, would return original
+ * @param {AddressType[]} fromTypes - to limit the types of input address, optional
+ * @returns {string} - return `toType` specific format of address
+ */
+export const getAddress = (
+  address: string,
+  toType?: AddressType,
+  fromTypes?: AddressType[],
+): string => {
+  if (!validation.isString(address)) {
+    throw new Error(`${address} is not string`);
+  }
+  const zilAddr = new ZilAddress(address);
+
+  const validateType: AddressType[] =
+    fromTypes === undefined || fromTypes.length === 0 ? [] : fromTypes;
+  let total = 0;
+
+  total =
```

Comment:
Is this necessary? I don't think we need `fromTypes`? We just check the input string `address`. I suggest adding `static isValidAddressEncoding(address: string)` to `ZilAddress`. This code is a little bit hard to follow here. Additionally, keep in mind that `enum` types are not visible to JS consumers of this library; it will be hard for them to use `AddressType`.

Reply:
Try and remake this one